modules/promotion/src/PromotionStorage.php

```diff
@@ -24,8 +24,27 @@ public function loadValid(OrderTypeInterface $order_type, StoreInterface $store)
     if (empty($result)) {
       return [];
     }
+
+    $usages_query = $this->database->select('commerce_promotion_usage', 'cpu');
```

Comment:
The API doesn't allow for a bulk look up, and core does this as well within storage to provide a more performant query. Otherwise we'd have to do a SELECT for each promotion, and that's wasteful